homeassistant/components/mqtt/client.py

```diff
@@ -543,21 +653,22 @@ def _client_unsubscribe(topic: str) -> int:
             _raise_on_error(result)
             return mid
 
+        if self._is_active_subscription(topic):
+            # Other subscriptions on topic remaining - don't unsubscribe.
+            return
+        async with self._pending_subscriptions_lock:
```

Comment:
Add:

```
if topic in self._max_qos:
  del self._max_qos[topic]
```

If we are unsubscribing, there are no more subscribers to this topic, so should clean up the max_qos.